analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/EmptyCollectionsShouldNotBeEnumerated.cs

```diff
@@ -516,59 +516,64 @@ public void DeriveFromTrackedCollection()
-        list.Average();                                     // FN
+        list.Aggregate((x, y) => x + y);                    // Noncompliant
+        list.All(x => true);                                // Noncompliant
+        list.Any();                                         // Noncompliant
+        list.Average();                                     // Noncompliant
+        list.Contains(5, EqualityComparer<int>.Default);    // Noncompliant
+        list.Distinct();                                    // Noncompliant
+        list.ElementAt(0);                                  // Noncompliant
+        list.ElementAtOrDefault(0);                         // Noncompliant
+        list.Except(list);                                  // Noncompliant
+        list.First();                                       // Noncompliant
+        list.FirstOrDefault();                              // Noncompliant
+        list.GroupBy(x => x);                               // Noncompliant
+        list.GroupJoin(list, x => x, x => x, (x, y) => x);  // Noncompliant
+        list.Intersect(list);                               // Noncompliant
+        list.Last();                                        // Noncompliant
+        list.LastOrDefault();                               // Noncompliant
+        list.LongCount();                                   // Noncompliant
+        list.Max();                                         // Noncompliant
+        list.Min();                                         // Noncompliant
+        list.OrderBy(x => x);                               // Noncompliant
+        list.OrderByDescending(x => x);                     // Noncompliant
+        list.Reverse();                                     // Noncompliant
+        list.Select(x => x);                                // Noncompliant
+        list.SelectMany(x => new int[5]);                   // Noncompliant
+        list.SequenceEqual(list);                           // Noncompliant
+        list.Single();                                      // Noncompliant
+        list.SingleOrDefault();                             // Noncompliant
+        list.Skip(1);                                       // Noncompliant
+        list.SkipWhile(x => true);                          // Noncompliant
+        list.Sum();                                         // Noncompliant
+        list.Take(1);                                       // Noncompliant
+        list.TakeWhile(x => true);                          // Noncompliant
+        list.Where(x => true);                              // Noncompliant
+        list.Zip(list, (x, y) => x);                        // Noncompliant
+        Enumerable.Reverse(list);                           // Noncompliant
+        list.Clear();                                       // Noncompliant should raise, because the methods above should not reset the state
+
         list.Cast<byte>();                                  // FN
```

Comment:
Why is Cast and OfType skipped?

Reply:
They return false in the ProcessInvocation, due to this check:
`&& instance.Type.DerivesOrImplementsAny(TrackedCollectionTypes)) // To be improved to handle OfType and Cast FNs`

Reply:
Is there a reason not to add System.Collections.Enumerable to the tracked collection types?